NethermindEth/nethermind
# Do not throw IOException when file is already downloaded

Fixes Closes Resolves #6315 

## Changes

Added stages tracking to avoid following issues:
1. When client was trying to download snapshot when it was already downloaded, it went to infinite loop of IOExceptions
2. When we didn't finished extracting snapshot, we need to reastart it, not to treat it as extractred

## Types of changes

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Other: _Description_

## Testing

#### Requires testing

- [ x] Yes
- [] No

#### If yes, did you write tests?


#### Notes on testing

* Tested downloading and unzipping flows
  * Download->Interrupt->Continue downloading
  * Download->Interrupt unzipping->Redownload
  * Download->Wait for completion->Redownload->Ensure download is not started

## Documentation

#### Requires documentation update


_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes
